esphome/components/byte_bus/byte_bus.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "esphome/core/log.h"
+#include "esphome/core/gpio.h"
+
+namespace esphome {
+namespace byte_bus {
+
+/**
+ * A pin to replace those that don't exist.
+ */
+class NullPin : public GPIOPin {
+ public:
+  void setup() override {}
+
+  void pin_mode(gpio::Flags _) override {}
+
+  bool digital_read() override { return false; }
+
+  void digital_write(bool _) override {}
+
+  std::string dump_summary() const override { return {"Not used"}; }
+};
+
+// https://bugs.llvm.org/show_bug.cgi?id=48040
+static GPIOPin *const NULL_PIN = new NullPin();  // NOLINT(cppcoreguidelines-avoid-non-const-global-variables)
+
+/**
+ * A class that can write a byte-oriented bus interface with CS and DC controls, typically used for
+ * LCD display controller interfacing.
+ */
+class ByteBus {
```

Comment:
```suggestion
class DataBus {
```
I suggest to make it more widely usable like for 16 bit mode etc.

Reply:
No problem with that.